micrometer-java11/src/test/java/io/micrometer/java11/instrument/binder/jdk/MicrometerHttpClientTests.java

```diff
@@ -91,6 +99,25 @@ void shouldInstrumentHttpClientWithTimer(WireMockRuntimeInfo wmInfo) throws IOEx
         thenMeterRegistryContainsHttpClientTags();
     }
 
+    @Test
+    @Issue("#5136")
+    void shouldThrowErrorFromSendAsync(WireMockRuntimeInfo wmInfo) {
+        var client = MicrometerHttpClient.instrumentationBuilder(httpClient, meterRegistry).build();
+
+        var request = HttpRequest.newBuilder(URI.create(wmInfo.getHttpBaseUrl() + "/test-fault")).GET().build();
+
+        var response = client.sendAsync(request, HttpResponse.BodyHandlers.ofString());
+
+        assertThatThrownBy(response::join).isInstanceOf(CompletionException.class);
+        assertThatNoException().isThrownBy(() -> meterRegistry.get("http.client.requests")
+            .tag("method", "GET")
+            // TODO fix status/uri in exceptional cases where response may be null
```

Comment:
This is not a TODO we should tackle in this PR. This should probably be fixed on `main` instead since it results in a change in behavior. Ideally, we should add more to our `HttpClientTimingInstrumentationVerificationTests` to ensure consistent behavior across instrumentations. The issue, if I remember correctly, is that some HTTP clients cannot instrument the case where the response is null.